sorc/orog_mask_tools.fd/lake.fd/lakefrac.F90

```diff
@@ -660,53 +743,45 @@ SUBROUTINE write_lakedata_to_orodata(cs_res, cs_lakestat, cs_lakedpth)
       lake_frac (:) = cs_lakestat ((tile_num-1)*tile_sz+1:tile_num*tile_sz)
       lake_depth(:) = cs_lakedepth((tile_num-1)*tile_sz+1:tile_num*tile_sz)
 
-! add Caspian Sea and Aral Sea to lake_frac and lake_depth
-      IF (tile_num == 2 .or. tile_num == 3) THEN
-        DO i = 1, tile_sz
-          IF (land_frac(i) < 0.9 .AND. lake_frac(i) < 0.1) THEN
-            IF (geolat(i) > 35.0 .AND. geolat(i) <= 50.0 .AND. &
-                geolon(i) > 45.0 .AND. geolon(i) <= 55.0) THEN
-              lake_frac(i) = 1.-land_frac(i)
-              lake_depth(i) = 211.0
-            ENDIF
-            IF (geolat(i) > 35.0 .AND. geolat(i) <= 50.0 .AND. &
-                geolon(i) > 57.0 .AND. geolon(i) <= 63.0) THEN
-              lake_frac(i) = 1.-land_frac(i)
-              lake_depth(i) = 10.0
-            ENDIF
-          ENDIF
-        ENDDO
-      ENDIF
-
-! adjust land_frac and lake_frac, and make sure land_frac+lake_frac=1 at inland points
-      DO i = 1, tile_sz
-        if (lake_frac(i) >= lake_cutoff) then ! non-zero lake_frac dominates over land_frac
-          land_frac(i) = max(0., min(1., 1.-lake_frac(i)))
-        elseif (inland(i) == 1.) then ! land_frac dominates over lake_frac at inland points
-          lake_frac(i) = max(0., min(1., 1.-land_frac(i)))
-        end if
+! include Caspian Sea and Aral Sea if GLDB data set is used, and 
+! exclude lakes in the coastal areas of Antarctica if MODIS data set is used  
+      CALL include_exclude_lakes(lake_frac,land_frac,lake_depth,geolat,geolon,tile_num)
 
 ! epsil is "numerical" nonzero min for lake_frac/land_frac
 ! lake_cutoff/land_cutoff is practical min for lake_frac/land_frac
-        if (min(lake_cutoff,land_cutoff) < epsil) then
-          print *,'lake_cutoff/land_cutoff cannot be smaller than epsil, reset...'
-          lake_cutoff=max(epsil,lake_cutoff)
-          land_cutoff=max(epsil,land_cutoff)
-        end if
+      IF (min(lake_cutoff,land_cutoff) < epsil) then
+        PRINT *,'lake_cutoff/land_cutoff cannot be smaller than epsil, reset...'
+        lake_cutoff=max(epsil,lake_cutoff)
+        land_cutoff=max(epsil,land_cutoff)
+      ENDIF
 
-        if (lake_frac(i)<   lake_cutoff) lake_frac(i)=0.
-        if (lake_frac(i)>1.-lake_cutoff) lake_frac(i)=1.
-        if (inland(i) == 1.) land_frac(i) = 1.-lake_frac(i)
+! adjust land_frac and lake_frac, and make sure land_frac+lake_frac=1 at inland points
+      DO i = 1, tile_sz
         if (land_frac(i)<   land_cutoff) land_frac(i)=0.
         if (land_frac(i)>1.-land_cutoff) land_frac(i)=1.
 
+        if (inland(i) /= 1.) then 
+          lake_frac(i) = 0.
+        endif
+
+        if (lake_frac(i) < lake_cutoff) lake_frac(i)=0.
```

Comment:
add this here:
if (lake_frac(i) >= lake_cutoff .and. gridcell_lake == 1) lake_frac(i)=1.0